app/models/auction.rb

```diff
@@ -62,6 +64,12 @@ def lowest_bids
     bids.select { |b| b.amount == lowest_amount }.sort_by(&:created_at)
   end
 
+  def self.find_by_token(token)
+    a = where(token: token).first
+    fail ActiveRecord::RecordNotFound if a.nil?
```

Comment:
I think that is what rails will do for us if we just do `find_by(token:token` above